resolwe/flow/models/utils.py

```diff
@@ -747,3 +747,68 @@ def referenced_files(data, include_descriptor=True):
     )
 
     return special_files + output_files + descriptor_files
+
+
+def get_collection_of_input_entities(data):
+    """Get collection that contains all "entity inputs" of a given data.
+
+    With "entity input", one refers to the inputs that are part of an entity.
+    """
+    # Prevent circular imports:
+    from resolwe.flow.models import Data, Collection
+
+    collections = set()
+
+    for field_schema, fields in iterate_fields(data.input, data.process.input_schema):
+        name = field_schema["name"]
+        value = fields[name]
+        if "type" not in field_schema:
+            continue
+        if field_schema["type"].startswith("data:"):
+            if value is None:
+                continue
+            try:
+                data = Data.objects.get(id=value)
+                if data.entity and data.collection:
+                    collections.add(data.collection.id)
+            except Data.DoesNotExist:
+                continue
+        elif field_schema["type"].startswith("list:data:"):
+            for val in value:
+                if val is None:
+                    continue
+                try:
+                    data = Data.objects.get(id=val)
+                    if data.entity and data.collection:
+                        collections.add(data.collection.id)
+                except Data.DoesNotExist:
+                    continue
+
+    if len(collections) != 1:
+        raise ValueError(
+            f"Entity inputs should be part of exactly one collection. (not {len(collections)})"
```

Comment:
We don't use f-strings in the codebase, so use `format` instead to keep it consistent.